dev/src/overmind/state.tsx

```diff
@@ -187,7 +207,96 @@ export const state: State = {
+            ? 1
+            : state.assignments[state.activeCourse].length ?? 0
+
+        let filteredSubmissions = submissions
+        for (const filter of state.submissionFilters) {
+            switch (filter) {
+                case "teachers":
+                    filteredSubmissions = filteredSubmissions.filter(submission => {
+                        return submission.enrollment ? !isTeacher(submission.enrollment) : false
+                    })
+                    break
+                case "approved":
+                    // approved filters all entries where all assignments have been approved
+                    filteredSubmissions = filteredSubmissions.filter(submission => {
+                        if (rootState.review.assignmentID > 0) {
+                            // If a specific assignment is selected, filter by that assignment
+                            const sub = submission.submissions?.find(submission => submission.getAssignment()?.getId() === rootState.review.assignmentID)?.getSubmission()
+                            return sub && isApproved(sub)
+                        }
+                        const numApproved = submission.submissions?.reduce((acc, cur) => {
+                            return acc + ((cur.hasSubmission() &&
+                                isApproved(cur.getSubmission() as Submission)) ? 1 : 0)
+                        }, 0) ?? 0
+                        return numApproved < numAssignments
+                    })
+                    break
+                default:
+                    break
+            }
+        }
+
+        const m = state.sortAscending ? -1 : 1
+        const sortedSubmissions = Object.values(filteredSubmissions).sort((a, b) => {
+            let subA: Submission | undefined
+            let subB: Submission | undefined
+            if (rootState.review.assignmentID > 0) {
+                // If a specific assignment is selected, sort by that assignment
+                subA = a.submissions?.find(submission => submission.getAssignment()?.getId() === rootState.review.assignmentID)?.getSubmission()
+                subB = b.submissions?.find(submission => submission.getAssignment()?.getId() === rootState.review.assignmentID)?.getSubmission()
+            }
+
+            switch (state.sortSubmissionsBy) {
+                case SubmissionSort.Score:
+                    if (rootState.review.assignmentID > 0) {
+                        const sA = subA?.getScore()
+                        const sB = subB?.getScore()
+                        if (sA !== undefined && sB !== undefined) {
+                            return m * (sB - sA)
+                        } else if (sA !== undefined) {
+                            return -m
+                        }
+                        return m
+                    }
+                    const aSubs = a.submissions?.reduce((acc, cur) => { return acc + (cur.hasSubmission() ? (cur.getSubmission() as Submission).getScore() : 0) }, 0) ?? 0
```

Comment:
Quite long line. And a bit tricky to see what's going on. Especially at the end, with many zero values. Same comment for Lines 287-288. Maybe a helper function can alleviate the problem.

Reply:
Agreed. Will add helpers for this and other areas you mention.